deeplearning4j-scaleout/spark/dl4j-spark/src/main/spark-1/org/deeplearning4j/spark/util/BaseDoubleFlatMapFunctionAdaptee.java

```diff
@@ -0,0 +1,25 @@
+package org.deeplearning4j.spark.util;
+
+import org.apache.spark.api.java.function.DoubleFlatMapFunction;
+
+import java.util.Iterator;
+
+/**
+ * DoubleFlatMapFunction adapter to hide incompatibilities between Spark 1.x and Spark 2.x
+ *
+ * This class should be used instead of direct referral to DoubleFlatMapFunction
+ *
+ */
+public class BaseDoubleFlatMapFunctionAdaptee<T> implements DoubleFlatMapFunction<T> {
```

Comment:
Ok done it, since we are control of FlatMapFunctionAdapter we could do it like this.